modules/data.remote/R/call_MODIS.R

```diff
@@ -27,65 +29,116 @@
 ##' 
 ##' @author Bailey Morrison
 ##'  
-call_MODIS <- function(outfolder = ".", start_date, end_date, lat, lon, size = 0, product, band, band_qc = "", band_sd = "", package_method = "MODISTools") {
+call_MODIS <- function(outfolder = NULL, start_date, end_date, lat, lon, size = 0, product, band, band_qc = "", band_sd = "", siteID = NULL, package_method = "MODISTools", QC_filter = F) {
   
   # makes the query search for 1 pixel and not for rasters for now. Will be changed when we provide raster output support.
   size <- 0
+# reformat start and end date if they are in YYYY/MM/DD format instead of YYYYJJJ
+  if (grepl("/", start_date) == T)
+  {
+    start_date = as.Date(paste0(lubridate::year(start_date), spatial.tools::add_leading_zeroes(lubridate::yday(start_date), 3)), format = "%Y%j")
```

Comment:
Unless I'm missing something here, I think you can eliminate the `spatial.tools` dependency here by using base R's `sprintf`:

```r
as.Date(sprintf("%04d%03d", lubridate::year(start_date), lubridate::yday(start_date)), format = "%Y%j")
```

The syntax is a bit confusing so, parsing out `%03d`:
- `%` indicates the beginning of a string substitution
- `0` means you'll be padding any extra width with zeros 
- `3` is the target string width
- `d` means it expects a number ("double")

Reply:
@bailsofhay agreed, no need for extra dependencies just to add leading 0s.  Make this change please